### PR TITLE
DOC: Add installation instructions for various platforms

### PR DESCRIPTION
**Pull request by shaneahmed**
- Add installation instructions for Windows, Linux and Mac

### CODE REVIEW

**Review by DavidBAEpstein — CHANGES_REQUESTED** [review]

File: docs/installation.rst [hunk 1]
```diff
@@ -5,6 +5,38 @@ Installation
...
+Linux (Ubuntu)
+^^^^^^^^^^^^^^
+On Linux the preprequiste software can be installed using the command
+
+.. code-block:: console
+
+    $ apt-get -y install libopenjp2-7-dev libopenjp2-tools openslide-tools
+
+This is the same command used in Google Colab examples. Although Google Colab examples remove `datascience`
+and `albumentations` due to compatibility issues.
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst:34]: Good point we should mention that the toolbox is compatible with python 3.6, 3.7 and 3.8

File: README.md [hunk 2]
```diff
@@ -52,6 +52,7 @@ The [bash](https://www.gnu.org/software/bash) shell is available on all commonly
 ### Install Python package
 
 If you wish to use our programs, perhaps without developing them further, run the command `pip install tiatoolbox`.
+The detailed installation instructions can be found in the [documentation](https://tia-toolbox.readthedocs.io/en/latest/installation.html).
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply README.md:55]: In example taster, the first step takes you to the examples page on Git. The second step follows the first step. I would assume a user is following all the steps and not skipping any.

[the remark this answers is not in the document]
Comment (shaneahmed) [reply README.md]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/f5968babb1fc21907d9d5c877306b0f8449f2e7a

File: docs/installation.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/f5968babb1fc21907d9d5c877306b0f8449f2e7a

**Comment by John-P** [comment]
It appears that different install methods are being mixed up a little here, which I found confusing. For example, windows says to install openslide with an installer but then openjpeg with conda and then use pip install for the remainder of the packages. On Ubuntu is says to use apt to install openslide and openjpeg, but you can install both of these via conda too (as with openjpeg in the windows section just before). While you can do this, conda recommends that you install everything through conda if you can.

I suggest a structure a bit like this for the instructions:

```
- Briefly outline install methods i.e.: conda, pip (+ system package-manager), docker (maybe, if we make a dockerfile?)
- Windows
  - conda (reccommended / least steps)
  - pip
    - System package prerequisites
- Linux (Ubuntu/Debian)
  - conda (reccommended / least steps)
  - pip
    - System package prerequisites
- macOS
  - conda (reccommended / least steps)
  - pip
    - System package prerequisites
      - homebrew
      - macports
```

Alternatively could do the 'recommended' method for each first then alternative methods in a different section below.

Also, @shaneahmed  I remember discussing that there is this channel which can install openslide for windows, linux and macos (the conda-forge one is linux and mac only). Did we ever test this out? That way we could use one conda file to install everything on each platform.

**Comment by shaneahmed** [comment]
> It appears that different install methods are being mixed up a little here, which I found confusing. For example, windows says to install openslide with an installer but then openjpeg with conda and then use pip install for the remainder of the packages. On Ubuntu is says to use apt to install openslide and openjpeg, but you can install both of these via conda too (as with openjpeg in the windows section just before). While you can do this, conda recommends that you install everything through conda if you can.
> 
> I suggest a structure a bit like this for the instructions:
> 
> ```
> - Briefly outline install methods i.e.: conda, pip (+ system package-manager), docker (maybe, if we make a dockerfile?)
> - Windows
>   - conda (reccommended / least steps)
>   - pip
>     - System package prerequisites
> - Linux (Ubuntu/Debian)
>   - conda (reccommended / least steps)
>   - pip
>     - System package prerequisites (apt)
> - macOS
>   - conda (reccommended / least steps)
>   - pip
>     - System package prerequisites
>       - homebrew
>       - macports
> ```
> 
> Alternatively could do the 'recommended' method for each first then alternative methods in a different section below.
> 
> Also, @shaneahmed I remember discussing that there is [this channel](https://anaconda.org/sdvillal/openslide) which can install openslide for windows, linux and macos (the conda-forge one is linux and mac only). Did we ever test this out? That way we could use one conda file to install everything on each platform.

There is no openslide channel for Windows on conda as far as I know. The best way is to download the binaries and add to system path.

**Comment by John-P** [comment]
@shaneahmed 
> There is no openslide channel for Windows on conda as far as I know. The best way is to download the binaries and add to system path.

[This one](https://anaconda.org/sdvillal/openslide) appears to be for windows Mac and Linux. I can give it a test and see if it works when I am next at home (I am in the office right now):
image

File: docs/installation.rst [hunk 3]
```diff
@@ -1,21 +1,94 @@
...
+
+
+Alternative Method
+==================
+
+If you cannot use Anaconda or are having trouble with it, you can try and alternative install method. Here we will show how to install pre-requisite binary packages before installing pythong dependencies using the Python package manager (pip).
+
+Windows
+^^^^^^^
+1. Download OpenSlide binaries from `this page <https://openslide.org/download/>`_ and add it to
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst:29]: There are several files but we need to add path to openslide directory. I will clarify this.

File: docs/installation.rst [hunk 4]
```diff
@@ -1,21 +1,94 @@
...
+Windows `system path <https://docs.microsoft.com/en-us/previous-versions/office/developer/sharepoint-2010/ee537574(v=office.14)>`_.
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst:30]: A directory containing files is added. I will clarify this.

File: docs/installation.rst [hunk 5]
```diff
@@ -1,21 +1,94 @@
...
+
+.. code-block:: console
+
+    $ conda env create --file requirements.conda.yml
+
+
+Alternative Method
+==================
+
+If you cannot use Anaconda or are having trouble with it, you can try and alternative install method. Here we will show how to install pre-requisite binary packages before installing pythong dependencies using the Python package manager (pip).
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/288dacaf430cb8637a1a91381c9dc46ad783b9be

File: docs/installation.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst:30]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/288dacaf430cb8637a1a91381c9dc46ad783b9be

File: docs/installation.rst [hunk 6]
```diff
@@ -1,21 +1,94 @@
...
+Linux (Ubuntu)
+^^^^^^^^^^^^^^
+On Linux the preprequiste software can be installed using the command
+
+.. code-block:: console
+
+    $ apt-get -y install libopenjp2-7-dev libopenjp2-tools openslide-tools
+
+The same command is used when working on the Colab or Kaggle platforms.
+In addition, we remove the packages `datascience` and `albumentations` from Colab because they conflict
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/288dacaf430cb8637a1a91381c9dc46ad783b9be

File: docs/installation.rst [hunk 7]
```diff
@@ -1,21 +1,94 @@
...
+2. Install OpenJPEG. The easiest way is to install OpenJpeg is through conda
+using
+
+.. code-block:: console
+
+    C:\> conda install -c conda-forge openjpeg>=2.3.0
+
+Linux (Ubuntu)
+^^^^^^^^^^^^^^
+On Linux the preprequiste software can be installed using the command
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply docs/installation.rst]: Addressed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/127/commits/288dacaf430cb8637a1a91381c9dc46ad783b9be

**Pull request merged**